File: backend/src/main/java/realty/domain/dto/MapInfo.java
package realty.domain.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/*
 * 수업명 : 가비아 2회차
 * 이름 : 김관호
 * 작성자 : 김관호
 * 수정자 : 
 * 작성일 : 25.07.30
 * 파일명 : MapInfo.java
 */

 // OCRResponse로 받을 맵에 대한 정보 dto

@Getter
@Setter
@NoArgsConstructor
public class MapInfo {

    // addresses
    private String address = "";       // 주소
    private String x = "";                 // X 좌표(경도)
    private String y = "";                 // Y 좌표(위도)

    // addressElements
    private String sido = "";              // 시/도
    private String sigugun = "";           // 시/구/군
    private String dongmyun = "";          // 동/면
    private String buildingName = "";      // 건물 이름

    @Override
    public String toString() {
        return "주소: " + address + "\n" +
               "위도: " + y + "\n" +
               "경도: " + x + "\n" +
               "시도: " + sido + "\n" +
               "시군구: " + sigugun + "\n" +
               "동/면: " + dongmyun + "\n" +
               "건물명: " + buildingName;
    }
}

File: ai-api/api/ocr/data/MapInfo.py
from dataclasses import dataclass


@dataclass
class MapInfo:
    #addresses
    address: str=""             #주소
    x : str = ""                #X 좌표(경도)            
    y : str = ""                #Y 좌표(위도)

    #addressElements
    sido : str = ""             #시/도
    sigugun : str = ""          #시/구/군
    dongmyun : str = ""         #동/면
    buildingName : str = ""     #건물 이름

    def __str__(self):
        return (
            f"도로명 주소: {self.roadAddress}\n"
            f"지번 주소: {self.jibunAddress}\n"
            f"위도: {self.y}\n"
            f"경도: {self.x}\n"
            f"시도: {self.sido}\n"
            f"시군구: {self.sigugun}\n"
            f"동/면: {self.dongmyun}\n"
            f"리: {self.ri}\n"
            f"건물명: {self.buildingName}\n"
        )
    